### PR TITLE
prepare publication

### PR DESCRIPTION
**Pull request by ranocha**
Since we're adding more features to this repo, it's getting more interest from others. Thus, it would be nice to polish it a bit - so that we can also prepare a software paper in the future. I have started this process in this PR.

@JoshuaLampert If you're fine with this, could you maybe list some breaking changes in the NEWS.md file?

### CODE REVIEW

**Comment by JoshuaLampert** [comment]
Thanks a lot for initiating that! I will fill the NEWS.md file with the changes from the last releases.

**Comment by JoshuaLampert** [comment]
> @JoshuaLampert If you're fine with this, could you maybe list some breaking changes in the NEWS.md file?

I filled the NEWS.md file. Please have a look if this looks fine to you.

**Review by ranocha — COMMENTED** [review]
Nice, thanks 👍

**Comment by ranocha** [comment]
Let's see how the rendered changelog looks in the docs. If it's fine, I'm fine merging this PR

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+https://www.contributor-covenant.org/version/2/0/code_of_conduct.html.
+
+Community Impact Guidelines were inspired by [Mozilla's code of conduct
+enforcement ladder](https://github.com/mozilla/diversity).
+
+[homepage]: https://www.contributor-covenant.org
+
+For answers to common questions about this code of conduct, see the FAQ at
+https://www.contributor-covenant.org/faq. Translations are available at
+https://www.contributor-covenant.org/translations.
```
Comment (JoshuaLampert) [inline CODE_OF_CONDUCT.md:129]: These links are not rendered nicely.

Comment (JoshuaLampert) [reply CODE_OF_CONDUCT.md:129]: Especially the links a few lines above.

Comment (JoshuaLampert) [reply CODE_OF_CONDUCT.md:129]: And they are not clickable.

**Comment by JoshuaLampert** [comment]
The changelog looks nice, I think.

**Comment by ranocha** [comment]
I adapted the text in the code of conduct so that the links are better integrated in the markdown text.

**Comment by ranocha** [comment]
> The changelog looks nice, I think.

I agree - nice work 👍

**Review by JoshuaLampert — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**